frontend/src/components/hooks/usePlaces.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+import { addPlace, getPlaces, updatePlace } from '../../service/PlaceService';
+
+export default function usePlaces() {
+  const [places, setPlaces] = useState([]);
+
+  useEffect(() => {
+    getPlaces().then(setPlaces);
+  }, []);
+
+  const create = (
+    primaryPictureUrl,
+    type,
+    title,
+    street,
+    address,
+    latitude,
+    longitude,
+    placeDescription,
+    pictureDescription,
+    aperture,
+    focalLength,
+    shutterSpeed,
+    iso,
+    flash,
+    youTubeUrl,
+    extraOne,
+    extraTwo,
+    particularities
+  ) =>
+    addPlace(
+      primaryPictureUrl,
+      type,
+      title,
+      street,
+      address,
+      latitude,
+      longitude,
+      placeDescription,
+      pictureDescription,
+      aperture,
+      focalLength,
+      shutterSpeed,
+      iso,
+      flash,
+      youTubeUrl,
+      extraOne,
+      extraTwo,
+      particularities
+    ).then((newPlace) => setPlaces([...places, newPlace]));
+
+  const update = (
+    id,
+    primaryPictureUrl,
+    type,
+    title,
+    street,
+    address,
+    latitude,
+    longitude,
+    placeDescription,
+    pictureDescription,
+    aperture,
+    focalLength,
+    shutterSpeed,
+    iso,
+    flash,
+    youTubeUrl,
+    extraOne,
+    extraTwo,
+    particularities
```

Comment:
If you need to have a list or arguments, think about the rest operator
```
const update = (...params) => updatePlace(...params).then(...
```

Reply:
sounds new to me, i will look into it.